src/microbe_stage/PatchMapGenerator.cs

```diff
@@ -8,63 +10,386 @@
 [SuppressMessage("ReSharper", "StringLiteralTypo", Justification = "Patch names aren't proper words")]
 public static class PatchMapGenerator
 {
+    // IDs for the patches in the predefined map (not the IDs of patches in the procedural map!)
+    private static readonly int CoastalId = 0;
+    private static readonly int EstuaryId = 1;
+    private static readonly int TidepoolId = 2;
+    private static readonly int EpipelagicId = 3;
+    private static readonly int MesopelagicId = 4;
+    private static readonly int BathypelagicId = 5;
+    private static readonly int AbyssopelagicId = 6;
+    private static readonly int SeafloorId = 7;
+    private static readonly int CaveId = 8;
+    private static readonly int IceShelfId = 9;
+    private static readonly int VentsId = 10;
+
     public static PatchMap Generate(WorldGenerationSettings settings, Species defaultSpecies, Random? random = null)
     {
         var map = new PatchMap();
-
+        var predefinedMap = new PatchMap();
         random ??= new Random(settings.Seed);
 
+        // Return the classic map if settings require it, otherwise use it to draw the procedural map
+        predefinedMap = PredefinedMap(predefinedMap, TranslationServer.Translate("PANGONIAN"));
```

Comment:
Is there a reason this is done like this? I'd really like to have the code be so that we don't create a unnecessary patch map object when it won't be used. Do we need a general method to grab the world generation settings into the map we will end up using to get around this?

Also the translation key should be "PANGONIAN_REGION_NAME" to be clear to translators.

Reply:
I think @Kemikal1 reused those patches generated in the predefined map. (Although I don't like this)

Reply:
Thats correct. If you wanted to make a certain each time look at the code for generating a default map, now duplicate it 2-3 times. almost 1000 lines of code.

Reply:
What I don't get is why the generation code even needs to run for the default patches.
I still haven't looked through the code but I imagine it would make total sense to have the map property setting as a separate method that could be passed the finished classic map, which should be hard coded in terms of the regions etc. it contains, and be done with it. Basically that way there is no code duplication and no need to have multiple map objects be created.

Reply:
I refactored this.